### PR TITLE
udf: parse an url to get host, path, query, port, protocol

### PR DESCRIPTION
**Pull request by xinghuayu007**
## Proposed changes

This is an user define function to parse an url to gt host, path, query, port and protocol.

The issue: #4320 

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [] I have create an issue on #ISSUE, and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [] If this change need a document change, I have updated the document
- [] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: contrib/udf/src/udf_samples/parse_url.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "udf_sample.h"
+#include <string>
+#include<algorithm>
```
Comment (WingsGo) [inline contrib/udf/src/udf_samples/parse_url.cpp:20]: #include <algorithm>

**Comment by WingsGo** [comment]
Hi, @xinghuayu007 Can you create an issue to explain why you add the udf parse url?Thx~~

File: contrib/udf/src/udf_samples/parse_url.cpp (hunk 1 above)
Comment (xinghuayu007) [reply contrib/udf/src/udf_samples/parse_url.cpp]: fixed

**Comment by xinghuayu007** [comment]
> Hi, @xinghuayu007 Can you create an issue to explain why you add the udf parse url?Thx~~

add issue #4320

**Comment by WingsGo** [comment]
> > Hi, @xinghuayu007 Can you create an issue to explain why you add the udf parse url?Thx~~
> 
> add issue #4320

I think If you want add a udf to Doris, you shouldn't add it in udf_samples, it is just for developer to create their own udf, it is not universal, you should add the udf inside Doris, maybe you can reference the implement of inline function such as `from_unixtime`

**Pull request closed without merging**